pkg/monitor.go

```diff
@@ -78,13 +78,17 @@ type PoolStats struct {
 // PoolsStats alias for map[string]PoolStats
 type PoolsStats map[string]PoolStats
 
+// Types of workloads can be deployed depending on the network manager running on the node
+type NodeFeature string
+
 // SystemMonitor interface (provided by noded)
 type SystemMonitor interface {
 	NodeID() uint32
 	Memory(ctx context.Context) <-chan VirtualMemoryStat
 	CPU(ctx context.Context) <-chan TimesStat
 	Disks(ctx context.Context) <-chan DisksIOCountersStat
 	Nics(ctx context.Context) <-chan NicsIOCounterStat
+	GetNodeFeatures() ([]NodeFeature, error)
```

Comment:
About the suggested interface, will this ever fail? worst case would be an empty slice :)